units/systemd-quotacheck-root.service.in

```diff
@@ -0,0 +1,26 @@
+#  SPDX-License-Identifier: LGPL-2.1-or-later
+#
+#  This file is part of systemd.
+#
+#  systemd is free software; you can redistribute it and/or modify it
+#  under the terms of the GNU Lesser General Public License as published by
+#  the Free Software Foundation; either version 2.1 of the License, or
+#  (at your option) any later version.
+
+[Unit]
+Description=File System Quota Check
+Documentation=man:systemd-quotacheck.service(8)
+
+ConditionPathExists={{QUOTACHECK}}
+ConditionPathExists=!/etc/initrd-release
+
+DefaultDependencies=no
+After=systemd-remount-fs.service
+Before=systemd-journal-flush.service shutdown.target
```

Comment:
why the journal ordering?

Reply:
(if this has merit, it should really be placed in the s-j-f.s, not here. and if this has merit, isn't it rather quotacheck-root that should have the dep?

Reply:
This is a bit tricky, because quotacheck should run when the fs is rw for creating the quota files (if necessary), but before anything is writing to it, because quotacheck mounts the fs temporarily ro.
My best bet to do this was before the journal-flush service as it seems this is the first one writing to the fs.
Of course, I'm happy to change that if you have an better idea.

Reply:
placed it in  s-j-f.s unit